app/routes.py

```diff
@@ -0,0 +1,337 @@
+    new_rental_response = None
+    
+    validate_video_id = Video.query.get_or_404(request_body["video_id"])
+    validate_customer_id = Customer.query.get_or_404(request_body["customer_id"])
+    validate_inventory = get_available_inventory(request_body)
+    if validate_inventory <= 0:
+        return jsonify({"message": "Could not perform checkout"}), 400
+
+    if validate_video_id and validate_customer_id and validate_inventory:
+        new_rental = Rental(customer_id=request_body["customer_id"],
+                        video_id=request_body["video_id"],
+                        due_date=due
+
+        )
+
+        db.session.add(new_rental)
+        db.session.commit()
+
+        new_rental_response = new_rental.to_dict()
+
+        rentals = Rental.query.filter_by(customer_id=request_body["customer_id"])
+    
+        videos_checked_out_count = rentals.count()
+        new_rental_response["videos_checked_out_count"] = videos_checked_out_count
+
+        after_check_out_inventory = get_available_inventory(request_body)
+        
+        new_rental_response["available_inventory"] = after_check_out_inventory
+
+        videos_in_rental = Rental.query.filter_by(customer_id=request_body["customer_id"])    
+        videos_checked_out_count = videos_in_rental.count()
+        new_rental_response["videos_checked_out_count"] = videos_checked_out_count
+        return new_rental_response
+
+    return jsonify(new_rental_response), 200
+
+# POST/Check-in a rental
+@rentals_bp.route("/check-in", methods = ["POST"])
+def update_rentals():
+    request_body = request.get_json()
+    if "video_id" not in request_body or "customer_id" not in request_body:
+        return "Bad data", 400
+
+    validate_video_id = Video.query.get_or_404(request_body["video_id"])
+    validate_customer_id = Customer.query.get_or_404(request_body["customer_id"])
+    if validate_video_id and validate_customer_id:
+        rentals = Rental.query.filter_by(customer_id=request_body["customer_id"], video_id=request_body["video_id"])
+        if not rentals.first():
+            return make_response({"message": "No outstanding rentals for customer 1 and video 1"}, 400)
+
+        db.session.delete(rentals.first())
+        db.session.commit()
+            
+        videos_in_rental = Rental.query.filter_by(customer_id=request_body["customer_id"])    
+        videos_checked_out_count = videos_in_rental.count()
+    
+        video = Video.query.get(request_body["video_id"])
+        video_in_rentals = Rental.query.filter_by(video_id=request_body["video_id"])
+
+        available_inventory = video.total_inventory - video_in_rentals.count()
+        if available_inventory < 0:
+            return "Could not perform checkout", 400
+    
+    return {
+        "customer_id": request_body["customer_id"],
+        "video_id": request_body["video_id"],
+        "videos_checked_out_count": videos_checked_out_count,
+        "available_inventory": available_inventory
+
+    }, 200
```

Comment:
This kind of dictionary would make a great method in a Rental model class.

Reply:
We thought that might be a way to refactor this part of the code. Thanks for confirming!